# AE 01: Getting Started

- Author: Your name here

**Important**

- For this exercise, you will work in groups, but everyone will work through the document and submit their solution.

- Make a duplicate copy of this project in your own Deepnote workspace.

- To submit this Activity, click the **Share** button in the top right hand corner of the screen, select "Edit" under "Anyone with a link to this project", and then copy the link into the corresponding Canvas assignment.

# Getting Started with R, Jupyter Notebooks, and Deepnote

The thing you are reading right now is a Jupyter Notebook. Jupyter notebooks can run several different programming languages, including R. Deepnote is an online collaborative environment built around notebooks — it lets you edit, run, and share notebooks with teammates. You can think of it like Google Docs for Data Science.


When you run blocks in a notebook, the output produced by code blocks appears immediately below the block. In Deepnote or other cloud notebook services you can run individual blocks (the ▶ button in the top righthand corner of each block) or use **Run** button at the top of the notebook to execute the entire notebook from top to bottom.

**All of your activities and homework assignments will be prepared using Jupyter Notebooks (.ipynb) and Deepnote.** You will submit them by sharing a link in the corresponding Canvas assignment and make sure to grant Edit privileges so I can access and comment on your work.

The nice thing about notebooks is that you can write prose (as I am doing now), mathematical equations using something called LaTeX (e.g. `$y_i = a + b x_i$`), and R code/output/plots all in one synthesized document. This makes it approximately 10,000 times easier to use than doing the same thing in Word or LaTeX. In the future, you'll often be given a notebook template to fill out. Make sure you add yourself as the author at the top and, if you're working in a group, add your teammates as well.

### Exercise 1

Run the notebook (or the block below) and show Dr. Friedlander before moving on.

## Using R as a Calculator and Running Code within a Notebook

With R by your side, you will never need your TI-84 again. Consider the simple arithmetic in the block below. You can run this code block all at once by clicking the ▶ button in the top righthand corner of the block. Notice the interactive nature of the output.

**As you work through this document, you should run each block as you come to it.**

### Exercise 2

Run the code block below.

In [3]:
# Insert code here
5 + 3

5.3 * 23.4

sqrt(16) # this is a comment.  R will not 'see' anything past a #, but you can use it to explain your code

[1] 8

[1] 124.02

[1] 4

Look closely at how Jupyter denotes the R code and the output. Also note how this notebook keeps R code separate from prose. These code blocks are called "cells" or "blocks". The easiest way to add a new block is to click the "..." menu button in the block above the one you want. Then, navigate down the menu or use the search function to find "Add code block below" if you want to insert a new code block or "Add markdown block below" if you want to write prose.

### Exercise 3

Create an R code block and compute the log of 10 using the function `log`. Why do you think the answer isn't 1?

In [4]:
# Insert code here
log(10)

[1] 2.302585

Insert interpretation here

### Exercise 4

You can also save values to named variables, to be used later:

In [5]:
# Insert code here
product <- 15.3 * 23.4 #save result

In [6]:
# Insert code here
product #show result

[1] 358.02

The symbol "\<-" is the *assignment operator*. Imagine that R maintains a warehouse filled with boxes containing all kinds of stuff. When you use "\<-", R will create a new box, write whatever name you put to the left of "\<-", and put whatever is to the right of "\<-" in the box. Then whenever you want to go get the box, you can just use whatever name you gave it. In the example above, the box is called "product" and it contains the number 358.02 (the result of 15.3 times 23.4). When you simple execute the command "product", R will go grab the box called "product" and dump it onto the screen. In the future, we will be giving names to entire datasets and using "functions" in R to analyze them.

If you've ever programmed before, "=" is the most common assignment operator. In R, you can use "=" as the assignment operator but there are cases where "\<-" works and "=" doesn't, so it's good to get into the habit of using "\<-" now.

Once variables are defined, they can be referenced with other operators and functions. Try executing the block below by clicking on it and pressing *Ctrl+Enter* (*Cmd +Enter* for Mac users).

In [7]:
# Insert code here
.5 * product # half of the product
product^(1/4) # fourth root of the product

[1] 179.01

[1] 4.349875

### Exercise 5

Run the block below three times and note what the output is. Then, use the Run button at the top to run the whole document and note what the output is. Are they the same?

In [ ]:
# Insert code here
product <- product * 2
product

<div style="border:2px solid #b71c1c; background-color:#ffebee; padding:1em; border-radius:6px; color:#212121;">
<strong>Important</strong>

When you click the run button for a block, it simply runs the code in the current session. I.e. as you make changes to R's warehouse, the code block you run will reference the current state of the warehouse. When you Run the whole notebook runs from the top it cleans out that warehouse and starts from the top. For this reason, if you run blocks multiple times or out of order, you may get unexpected results when you Run the whole document. Before you submit your documents, <strong>always restart your session and Run All and then double check your answers to make sure that your answers match your explanations.</strong>
</div>